objects/src/notes/origin.rs

```diff
@@ -0,0 +1,71 @@
+use super::{Digest, Felt, NoteError, NOTE_TREE_DEPTH};
+use crypto::merkle::{MerklePath, MmrProof, NodeIndex};
+
+/// Represents the origin of a note.  This includes the block number the note was created in, and
+/// the node index of the note in the note Merkle tree of that block.  The note origin is used by
+/// the transaction kernel to verify the authenticity of a note.
+#[derive(Debug)]
+pub struct NoteOrigin {
+    block_num: Felt,
+    sub_hash: Digest,
+    note_root: Digest,
+    node_index: NodeIndex,
+    chain_path: MmrProof,
+    note_path: MerklePath,
+}
```

Comment:
I'm not sure about including `chain_path` here. The main reason is that all elements here except for the chain path are stable - i.e., the don't change over time. But chain path will need to be re-computed from time to time.

Depending on the use case, I'm thinking chain path could be derived in different ways:
- For notes stored in user wallets, there will probably a "partial MMR" which keeps track of all relevant paths and updates them as needed.
- For notes stored by nodes, chain paths could come directly from the full MMR (which would likely be loaded into memory anyways).

Reply:
This makes sense - I'll remove it.